test/python/modules/linear_boltzmann_solvers/transport_td/transport_3d_2g_inf_med_pydrvr.py

```diff
@@ -0,0 +1,152 @@
+
+if __name__ == "__main__":
+
+    meshgen = FromFileMeshGenerator(filename="../../../../assets/mesh/cube3.2.msh")
+    grid = meshgen.Execute()
+    grid.SetUniformBlockID(0)
+    grid.SetOrthogonalBoundaries()
+
+    num_groups = 2
+    xs_diag = MultiGroupXS()
+    xs_diag.LoadFromOpenSn("simple_2g_downscatter_td.cxs")
+
+    # Total source in group 0, converted to volumetric rate
+    Q_tot = 122.58
+    Q_vol = Q_tot / (3.2 * 3.2 * 3.2)
+
+    strength = [0.0 for _ in range(num_groups)]
+    strength[0] = Q_vol  # source only in group 0
+    strength[1] = 0.0
+
+    # Volumetric source is effectively always on
+    mg_src = VolumetricSource(block_ids=[0], group_strength=strength)
+
+    # Angular quadrature
+    pquad = GLCProductQuadrature3DXYZ(n_polar=4, n_azimuthal=16, scattering_order=0)
+
+    gs0 = [0, num_groups - 1]
+
+    phys = DiscreteOrdinatesProblem(
+        mesh=grid,
+        num_groups=num_groups,
+        groupsets=[
+            {
+                "groups_from_to": gs0,
+                "angular_quadrature": pquad,
+                "angle_aggregation_type": "single",
+                "angle_aggregation_num_subsets": 1,
+                "inner_linear_method": "petsc_richardson",
+                "l_abs_tol": 1.0e-6,
+                "l_max_its": 500,
+            },
+        ],
+        xs_map=[
+            {"block_ids": [0], "xs": xs_diag},
+        ],
+        volumetric_sources=[mg_src],
+        boundary_conditions=[
+            {"name": "xmin", "type": "reflecting"},
+            {"name": "xmax", "type": "reflecting"},
+            {"name": "ymin", "type": "reflecting"},
+            {"name": "ymax", "type": "reflecting"},
+            {"name": "zmin", "type": "reflecting"},
+            {"name": "zmax", "type": "reflecting"},
+        ],
+        options={
+            "verbose_inner_iterations": False,
+        },
+        time_dependent=True,
+    )
+
+    # Create the time-dependent solver without stop_time, we will loop in Python
+    solver = TimeDependentSourceSolver(problem=phys, verbose=False)
+    solver.Initialize()
+
+    # Time stepping parameters
+    dt = 0.05
+    theta = 0.5
+    stop_time = 2.0
+
+    current_time = 0.0
+    step = 0
+
+    while current_time < stop_time:
+        target_time = min(current_time + dt, stop_time)
+        step_dt = target_time - current_time
+
+        # Configure the time step on the solver
+        solver.SetTimeStep(step_dt)
+        solver.SetTheta(theta)
+
+        if rank == 0:
+            print("")
+            print(
+                f"*************** Time step #{step:d}  t = {target_time:.6f} "
+                f"(from {current_time:.6f}, dt = {step_dt:.6f}, theta = {theta:.3f}) "
+                f"***************"
+            )
+
+        # Advance the solution from current_time to target_time
+        solver.Advance()
+
+        fflist = phys.GetScalarFieldFunctionList()
+        monitor_volume = RPPLogicalVolume(infx=True, infy=True, infz=True)
```

Comment:
should these be repeated inside the loop. If so, why?

Reply:
They should not. Fixes.